hal/src/main/native/mau/AnalogGyro.cpp

```diff
@@ -0,0 +1,144 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2016-2018 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "HAL/AnalogGyro.h"
+
+#include <chrono>
+#include <thread>
+#include <AHRS.h>
+
+#include "AnalogInternal.h"
+#include "HAL/AnalogAccumulator.h"
+#include "HAL/AnalogInput.h"
+#include "HAL/handles/IndexedHandleResource.h"
+#include "HALInitializer.h"
+#include "VMXPointers.h"
+#include <AHRS.h>
+
+using namespace hal;
+
+namespace hal {
+    struct AnalogGyro {
+        HAL_AnalogInputHandle handle;
+        uint8_t index;
+    };
+
+    static IndexedHandleResource<HAL_GyroHandle, AnalogGyro, kNumAccumulators,
+            HAL_HandleEnum::AnalogGyro>* analogGyroHandles;
+
+    namespace init {
+        void InitializeAnalogGyro() {
+            static IndexedHandleResource<HAL_GyroHandle, AnalogGyro, kNumAccumulators,
+                    HAL_HandleEnum::AnalogGyro> agH;
+            analogGyroHandles = &agH;
+        }
+    }
+}
+
+extern "C" {
+    HAL_GyroHandle HAL_InitializeAnalogGyro(HAL_AnalogInputHandle analogHandle, int32_t* status) {
+        hal::init::CheckInit();
+        if (!HAL_IsAccumulatorChannel(analogHandle, status)) {
+            if (*status == 0) {
+                *status = HAL_INVALID_ACCUMULATOR_CHANNEL;
+            }
+            return HAL_kInvalidHandle;
+        }
+
+        // handle known to be correct, so no need to type check
+        int16_t channel = getHandleIndex(analogHandle);
+
+        auto handle = analogGyroHandles->Allocate(channel, status);
+
+        if (*status != 0)
+            return HAL_kInvalidHandle;  // failed to allocate. Pass error back.
+
+        // Initialize port structure
+        auto gyro = analogGyroHandles->Get(handle);
+        if (gyro == nullptr) {  // would only error on thread issue
+            *status = HAL_HANDLE_ERROR;
+            return HAL_kInvalidHandle;
+        }
+
+        gyro->handle = analogHandle;
+        gyro->index = channel;
+
+    //    SimAnalogGyroData[channel].SetInitialized(true);
```

Comment:
The VMX-pi automatically initializes the singleton AHRS class when the main VMXPi constructor occurs.  So there is no initialization needed.